src/main/java/graphql/engine/EngineParameters.java

```diff
@@ -0,0 +1,184 @@
+package graphql.engine;
+
+import graphql.Assert;
+import graphql.Internal;
+import graphql.PublicApi;
+import graphql.execution.ExecutionStepInfo;
+import graphql.execution.MergedField;
+import graphql.execution.MergedSelectionSet;
+import graphql.execution.NonNullableFieldValidator;
+import graphql.execution.ResultPath;
+import graphql.language.OperationDefinition;
+
+import java.util.function.Consumer;
+
+import static graphql.Assert.assertNotNull;
+
+/**
+ * The parameters that are passed to {@link GraphQLEngine}s
+ */
+@PublicApi
+public class EngineParameters {
```

Comment:
This is pretty much a copy of ExecutionStrategyParameters but I felt that is too tied to the old ES way